### PR TITLE
MINIFICPP-1547 - Change default c2 protocol

### PR DESCRIPTION
**Pull request by adamdebreceni**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: C2.md [hunk 1]
```diff
@@ -63,11 +63,11 @@ be requested via C2 DESCRIBE manifest command.
 	# specify classes for the AST response
 	nifi.c2.root.classes=DeviceInfoNode,AgentInformation,FlowInformation
 	
-	# specify C2 protocol -- default is CoapProtocol if not specified
+	# specify C2 protocol -- default is RESTSender if not specified
+	nifi.c2.agent.protocol.class=RESTSender
 	#nifi.c2.agent.protocol.class=CoapProtocol
 	# may also use MQTT or REST
 	# nifi.c2.agent.protocol.class=MQTTC2Protocol
```
Comment (martinzink) [inline C2.md:70]: Thanks for the PR, I think we should also change the non-default help comments here as well to reflect the change
```suggestion
	# may also use MQTT or CoapProtocol
	# nifi.c2.agent.protocol.class=MQTTC2Protocol
	# nifi.c2.agent.protocol.class=CoapProtocol
```

**Review by martinzink — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

File: C2.md [hunk 2]
```diff
@@ -63,11 +63,11 @@ be requested via C2 DESCRIBE manifest command.
...
 	nifi.c2.root.classes=DeviceInfoNode,AgentInformation,FlowInformation
 	
-	# specify C2 protocol -- default is CoapProtocol if not specified
-	#nifi.c2.agent.protocol.class=CoapProtocol
-	# may also use MQTT or REST
-	# nifi.c2.agent.protocol.class=MQTTC2Protocol
+	# specify C2 protocol -- default is RESTSender if not specified
 	nifi.c2.agent.protocol.class=RESTSender
+	# may also use MQTT or CoapProtocol
+	# nifi.c2.agent.protocol.class=MQTTC2Protocol
```
Comment (arpadboda) [inline C2.md:69]: Do we have MQTT tested or at least documented anywhere?
Honestly, I have never tried it.

**Review by arpadboda — APPROVED** [review]

File: C2.md (hunk 2 above)
Comment (adamdebreceni) [reply C2.md:69]: I couldn't find it being tested anywhere, there is some sporadic mention of it in the docs, but nothing that would indicate that it has ever been used, we could investigate it in a separate PR if we would like to remove it, or revive it

**Review by szaszm — COMMENTED** [review]
I think we should also change the bootstrap.sh script to not enable COAP by default.
sed: `/^add_disabled_option COAP_ENABLED/s/TRUE/FALSE/`

**Review by szaszm — APPROVED** [review]

**Pull request closed without merging**